test_runner/fixtures/utils.py

```diff
@@ -613,3 +624,25 @@ def human_bytes(amt: float) -> str:
         amt = amt / 1024
 
     raise RuntimeError("unreachable")
+
+
+def all_pairs_component_versions() -> Iterable[object]:
+    """
+    This function generates all the pairs of old (False) or new (True)
+    versions of the Neon components
+    E.g. Pairs(storage_controller=False, storage_broker=True, compute=True, safekeeper=False, pageserver=False)
+    """
+    for pair in AllPairs(
+        OrderedDict({component: [True, False] for component in COMPONENT_BINARIES.keys()})
+    ):
```

Comment:
We need to add a constraint that there's no "all new" and no "all old" combinations

Reply:
But we need the "all-new" combination to test if it works on new versions

Reply:
Yeah, you're right, we need to ensure that combinations **must** include "all new", and **must not** include "all old"

Reply:
done